glslc/src/file_compiler.cc

```diff
@@ -96,8 +99,34 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
     return false;
   }
 
-  // Write output to output file.
-  output_stream->write(result.GetData(), result.GetLength());
+  // Get a string_piece which refers to the normal compilation output for now.
+  // This string_piece might be overwritten by dependency info dumping handler
+  // later, if the handler is instantiated to dump as normal compilation output,
+  // and the original compilation output should be blocked. Otherwise it won't
+  // be touched. The main output stream dump this string_piece later.
+  string_piece compilation_output(result.GetData(),
+                                  result.GetData() + result.GetLength());
+
+  // If we have dependency info dumping handler instantiated, we should dump
+  // dependency info first, this may overwrite the compilation output
```

Comment:
Just a minor wording fix to make it more clear:
Isn't it that the compilation_output might be redirected to reference the contents of this string?  That's not quite saying that compilation output string piece will be overwritten (The way you wrote it sounds like the *contents* of the string piece will be overwritten, i.e. the storage pointed to by the string piece.)